frame/lending/src/tests_offchain.rs

```diff
@@ -0,0 +1,159 @@
+};
+
+const DEFAULT_MARKET_VAULT_RESERVE: Perquintill = Perquintill::from_percent(10);
+
+#[test]
+fn test_liquidation_offchain_worker() {
+	let account_id = *ALICE;
+	let authority_id = wrapper::UintAuthorityIdWrapper::from(account_id);
+	wrapper::UintAuthorityIdWrapper::set_all_keys(vec![authority_id]);
+	// Create externalities, register transaction pool and key store in the externalities.
+	let mut ext = new_test_ext();
+	let (pool, pool_state) = testing::TestTransactionPoolExt::new();
+	ext.register_extension(TransactionPoolExt::new(pool));
+	ext.execute_with(|| {
+		let manager = *ALICE;
+		let lender = *CHARLIE;
+		// Create a market with BTC as collateral asset and USDT as borrow asset.
+		// Initial collateral asset price is 50_000 USDT. Market's collateral factor equals two.
+		// It means that borrow supposed to be undercolateraized when
+		// borrowed amount is higher then one half of collateral amount in terms of USDT.
+		let (market_id, vault) = create_market::<50_000>(
+			USDT::instance(),
+			BTC::instance(),
+			manager,
+			DEFAULT_MARKET_VAULT_RESERVE,
+			MoreThanOneFixedU128::saturating_from_integer::<u128>(2),
+		);
+
+		// Deposit collateral.
+		let collateral_value = BTC::units(1);
+		assert_ok!(Tokens::mint_into(BTC::ID, &manager, collateral_value));
+		assert_extrinsic_event::<Runtime>(
+			Lending::deposit_collateral(Origin::signed(manager), market_id, collateral_value),
+			Event::Lending(crate::Event::CollateralDeposited {
+				sender: manager,
+				amount: collateral_value,
+				market_id,
+			}),
+		);
+		// Deposit USDT in the vault.
+		let vault_value = USDT::units(100_000_000);
+		assert_ok!(Tokens::mint_into(USDT::ID, &lender, vault_value));
+		assert_ok!(Vault::deposit(Origin::signed(lender), vault, vault_value));
+		process_and_progress_blocks(1);
+		// Borrow 20_000 USDT.
+		let borrowed_value = USDT::units(20_000);
+		assert_extrinsic_event::<Runtime>(
+			Lending::borrow(Origin::signed(manager), market_id, borrowed_value),
+			Event::Lending(crate::Event::Borrowed {
+				sender: manager,
+				amount: borrowed_value,
+				market_id,
+			}),
+		);
+
+		// Emulate situation when collateral price has fallen down
+		// from 50_000 USDT to 38_000 USDT.
+		// Now the borrow is undercolateraized since market's collateral factor equals two.
+		// Therefore, one BTC can cover only 19_000 of 20_0000 borrowed USDT.
+		set_price(BTC::ID, NORMALIZED::units(38_000));
+		// Header for the fake block to execute off-chain worker
+		let header =
+			Header::new(2, H256::default(), H256::default(), [69u8; 32].into(), Digest::default());
+		// Execute off-chain worker
+		Executive::offchain_worker(&header);
+		// Check if corresponded liquidation transaction has been placed in the pool
+		let tx = pool_state.write().transactions.pop().unwrap();
+		assert!(pool_state.read().transactions.is_empty());
+		let tx = Extrinsic::decode(&mut &*tx).unwrap();
```

Comment:
not need in this pr, but as part of liquidation offchain testing, ensure that other positions not liquidated

Reply:
you may read on chain state to check it afaik. as off chain worker kind of can read, but not write, on chain state

Reply:
if you can check events, check that unhealthy positions liquidated and health retained

Reply:
>there are no events for healthy positions?
Liquidations pallet has only one event: PositionWasSentToLiquidation

https://github.com/ComposableFi/composable/blob/main/frame/liquidations/src/lib.rs#L116 

My point is that we do not achieve liquidation in our conditions. To liquidate something we need some activity from Liquidations side.